light-client/src/supervisor.rs

```diff
@@ -2,63 +2,49 @@ use crossbeam_channel as channel;
 
 use tendermint::evidence::{ConflictingHeadersEvidence, Evidence};
 
-use crate::{
-    bail,
-    callback::Callback,
-    errors::{Error, ErrorKind},
-    evidence::EvidenceReporter,
-    fork_detector::{Fork, ForkDetection, ForkDetector},
-    light_client::LightClient,
-    peer_list::PeerList,
-    state::State,
-    types::{Height, LightBlock, PeerId, Status},
-};
+use crate::bail;
+use crate::errors::{Error, ErrorKind};
+use crate::evidence::EvidenceReporter;
+use crate::fork_detector::{Fork, ForkDetection, ForkDetector};
+use crate::light_client::LightClient;
+use crate::peer_list::PeerList;
+use crate::state::State;
+use crate::types::{Height, LightBlock, PeerId, Status};
 
 pub trait Handle {
     /// Get latest trusted block from the [`Supervisor`].
-    fn latest_trusted(&mut self) -> Result<Option<LightBlock>, Error>;
+    fn latest_trusted(&self) -> Result<Option<LightBlock>, Error> {
+        todo!()
```

Comment:
Will this contain the default impl? The `todo!()` doesn't really communicate what is supposed to happen here.

Same with the other 3 functions below.

Reply:
The intention is that impls can overwrite the methods they depend on, but the default implementation should never be used without being overwritten. This way mocks and partial implementations are leaner, but can't be miss-used.